policy-controller/k8s/status/src/index.rs

```diff
@@ -74,14 +111,41 @@ impl Index {
         updates: UnboundedSender<Update>,
     ) -> SharedIndex {
         Arc::new(RwLock::new(Self {
-            _name: name.to_string(),
-            _claims: claims,
+            name: name.to_string(),
+            claims,
             updates,
             http_routes: HashMap::new(),
             servers: HashSet::new(),
         }))
     }
 
+    pub async fn run(index: Arc<RwLock<Self>>) {
+        // Clone the claims watch out of the index. This will immediately
+        // drop the read lock on the index so that it is not held for the
+        // lifetime of this function.
+        let mut claims = index.read().claims.clone();
+
+        loop {
+            tokio::select! {
+                _ = claims.changed() => {
+                    tracing::debug!("Lease holder has changed")
+                }
+                _ = tokio::time::sleep(std::time::Duration::from_secs(10)) => {}
+            }
+
+            // The claimant has changed or we should attempt to reconcile all
+            // HTTPRoutes to account for any errors. In either case, we should
+            // only proceed if we are the current leader.
+            let claims = claims.borrow_and_update();
+            let index = index.write();
```

Comment:
why is it necessary to acquire the write lock here? as far as I can tell, all the uses of `index` in this function (borrowing the `name` to check if we are the claimant, and then calling `Index::reconcile`, which takes `&self`) only need a read lock?

Reply:
Oh right good call. Yea not sure why I was thinking we needed `write` for this.